### PR TITLE
#642 Added config options maxLengthForMultiLineControls and maxLengthForSingleLineControls in propertiesConfig

### PR DESCRIPTION
**Pull request by nmgokhale**
Fixes https://github.com/elyra-ai/canvas/issues/642

Test harness -
Screen Shot 2021-05-04 at 4 07 23 PM
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-properties/properties-main/properties-main.jsx [hunk 1]
```diff
@@ -532,7 +532,8 @@ PropertiesMain.propTypes = {
 		schemaValidation: PropTypes.bool,
 		applyPropertiesWithoutEdit: PropTypes.bool,
 		conditionHiddenPropertyHandling: PropTypes.oneOf(["null", "undefined", "value"]),
-		conditionDisabledPropertyHandling: PropTypes.oneOf(["null", "undefined", "value"])
+		conditionDisabledPropertyHandling: PropTypes.oneOf(["null", "undefined", "value"]),
+		maxLength: PropTypes.number
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/properties-main/properties-main.jsx:536]: I think we might want 2 options.  One for multi-line controls like textarea or expressions and then another for textfield, date, etc... Right now we don't limit those but we might in the future.
We probably also want the ability to set a `-1` for no limit.

File: canvas_modules/common-canvas/src/common-properties/controls/textfield/textfield.jsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ class TextfieldControl extends React.Component {
 	constructor(props) {
 		super(props);
 		this.reactIntl = props.controller.getReactIntl();
-		this.charLimit = ControlUtils.getCharLimit(props.control, CHARACTER_LIMITS.TEXT_FIELD);
+		this.charLimit = ControlUtils.getCharLimit(props.control, props.controller.getPropertiesConfig().maxLength);
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/textfield/textfield.jsx:37]: Might be better to create a new controller method for getting these values so it's clearer

File: canvas_modules/common-canvas/src/common-properties/properties-main/properties-main.jsx (hunk 1 above)
Comment (nmgokhale) [reply canvas_modules/common-canvas/src/common-properties/properties-main/properties-main.jsx:536]: Adding 2 config options in `propertiesConfig`- 
```
maxLengthForMultiLineControls: PropTypes.number,
maxLengthForSingleLineControls: PropTypes.number
```

**Review by matthoward366 — APPROVED** [review]
Looks good.

**Pull request merged**